### ISSUE TITLE
Incorrect dead code elimination with class instanceof

### ISSUE THREAD

**Issue by AshleyScirra** [google/closure-compiler#3618]
Using Closure Compiler v20200517

in.js:
```js
window.Class = class MyClass {
	constructor()
	{
		if (this instanceof MyClass)
			console.log("test");
	}
}

new window.Class();
```
Note if you run this in a browser console it logs "test".

Compile with the following options:

`java -jar ./closure-compiler.jar --js in.js --js_output_file out.js --compilation_level ADVANCED --language_in ECMASCRIPT_2015 --language_out ECMASCRIPT_2015`

out.js looks like this:

```js
window.a=class{
	constructor(){}
};
new window.a;
```

Note it has incorrectly removed the console log. Note the class name has been removed, which may have then caused `this instanceof MyClass` to be identified as false and therefore dead code.

**Comment by concavelenz** [comment]
Thank you for the report and minimal repro case.  This look similiar to another issue @brad4d was looking and maybe has a fix pending for?

**Comment by brad4d** [comment]
This is similar to another issue that I fixed, but not quite the same.

I've reproduced the problem with the latest version of the compiler code by following the example here.

1. RemoveUnusedCode sees only one reference to `MyClass`, and it's an `instanceof` check, so it believes no objects of `MyClass` get created and replaces the `instanceof` check with `false`.
2. Then `RemoveUnusedCode` removes the `MyClass` name from the class definition.
